flang/test/Analysis/AliasAnalysis/alias-analysis-9.fir

```diff
@@ -17,13 +25,13 @@
 // end module
 
```

Comment:
Can we check aliasing between x%next and and y%next?

Reply:
Thanks for reviewing.  I've added that check.  It returns MayAlias.  As the comment I added says, I think it should return NoAlias.  We could try to fix that in this patch, but I think it would be better to address it in a separate patch because it involves a different part of the AliasAnalysis logic: the isTargetOrPointer() checks.

Reply:
I pushed the cases I wanted covered here: https://github.com/llvm/llvm-project/pull/95287

Reply:
Thanks.  I plan to take a look today.

Reply:
That part looks good to me.  Thanks.

Reply:
> Thanks for reviewing. I've added that check. It returns MayAlias. As the comment I added says, I think it should return NoAlias. We could try to fix that in this patch, but I think it would be better to address it in a separate patch because it involves a different part of the AliasAnalysis logic: the isTargetOrPointer() checks.

I decided to go ahead and [fix that](https://github.com/llvm/llvm-project/pull/94242/commits/ee248f5236ec4914cf8708167fd74f4b48c2733f) in this PR as it helped me to work through another issue.